File: schedule-service/src/main/java/pe/albrugroup/schedule_service/entity/response/asistencia/DetalleAsistenciaResponse.java
package pe.albrugroup.schedule_service.entity.response.asistencia;

import lombok.*;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.LocalTime;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class DetalleAsistenciaResponse {

    private Long idEmpleado;
    private Long idHorario;
    private LocalDate fecha;
    private LocalTime entradaProgramada;
    private LocalTime salidaProgramada;
    private LocalTime inicioAlmuerzoProgramado;
    private LocalTime finAlmuerzoProgramado;
    private LocalDateTime fechaHoraIngreso;
    private LocalDateTime fechaHoraSalida;
    private LocalDateTime fechaHoraInicioAlmuerzo;
    private LocalDateTime fechaHoraFinAlmuerzo;
    private Integer minutosObjetivoDia;
    private Integer minutosTrabajados;
    private Integer minutosBalance;
    private Integer minutosAlmuerzoTomados;
    private Integer minutosServiciosPermitidos;
    private Integer minutosServiciosAcumulados;
    private Boolean excedioServicios;
    private Boolean jornadaCerrada;
}

File: frontend/src/app/shared/models/schedule/detalle-asistencia-response.ts
import { EstadoAsistencia } from './estado-asistencia';

export interface DetalleAsistenciaResponse {
  idEmpleado: number;
  idHorario: number | null;
  fecha: string;
  estadoActual: EstadoAsistencia;
  entradaProgramada: string | null;
  salidaProgramada: string | null;
  inicioAlmuerzoProgramado: string | null;
  finAlmuerzoProgramado: string | null;
  fechaHoraIngreso: string | null;
  fechaHoraSalida: string | null;
  fechaHoraInicioAlmuerzo: string | null;
  fechaHoraFinAlmuerzo: string | null;
  minutosObjetivoDia: number | null;
  minutosTrabajados: number | null;
  minutosBalance: number | null;
  minutosAlmuerzoTomados: number | null;
  minutosServiciosPermitidos: number | null;
  minutosServiciosAcumulados: number | null;
  excedioServicios: boolean | null;
  jornadaCerrada: boolean;
}
